ndcube/ndcube.py

```diff
@@ -918,6 +919,37 @@ def __neg__(self):
         return self._new_instance(data=-self.data)
 
     def __add__(self, value):
+        if isinstance(value, NDData) and value.wcs is None:
+            if self.unit is not None and value.unit is not None:
+                    value_data = value.data * value.unit.to(self.unit)
+            elif self.unit is None:
+                value_data = value.data
+            else:
+                raise TypeError("Cannot add unitless NDData to a unitful NDCube.")
+
+            # addition
+            new_data = self.data + value_data
+            # combine the uncertainty
+            new_uncertainty = None
+            if self.uncertainty is not None and value.uncertainty is not None:
+                new_uncertainty = self.uncertainty.propagate(
+                    np.add, value.uncertainty, correlation=0
+                )
+            elif self.uncertainty is not None:
+                new_uncertainty = self.uncertainty
+            elif value.uncertainty is not None:
+                new_uncertainty = value.uncertainty
```

Comment:
Alternatively, you can use `else` to set `new_uncertainty` to `None`.
```suggestion
            if self.uncertainty is not None and value.uncertainty is not None:
                new_uncertainty = self.uncertainty.propagate(
                    np.add, value.uncertainty, correlation=0
                )
            elif self.uncertainty is not None:
                new_uncertainty = self.uncertainty
            elif value.uncertainty is not None:
                new_uncertainty = value.uncertainty
            else:
                new_uncertainty = None
```